Source/panels/spell_icons.hpp

```diff
@@ -11,6 +11,60 @@
 
 namespace devilution {
 
+enum class SpellIcon : uint8_t {
+	Firebolt,
+	Healing,
+	Lightning,
+	Flash,
+	Identify,
+	FireWall,
+	TownPortal,
+	StoneCurse,
+	Infravision,
+	HealOther,
+	Nova,
+	Fireball,
+	ManaShield,
+	FlameWave,
+	Inferno,
+	ChainLightning,
+	Sentinel, // unused
+	DoomSerpents,
+	BloodRitual,  // unused
+	Invisibility, // unused
+	Golem,
+	Etherealize,
+	BloodBoil,
+	Teleport,
+	Apocalypse,
+	ItemRepair,
+	None,

```

Comment:
None should be the "neutral" element in the enum (i.e. the first one, which is equal to 0).

Reply:
What should we call this frame then?

Reply:
Not sure what you mean. None should be called `None`, but it should be the first entry in the enum with value 0.

Then, on this list here, you'd put it in the correct spot based on it's underlying value (if the value matters, which it appears it does).

Reply:
What's a good solution

Reply:
`enum class SpellIcon : uint8_t {
	None = 26,
	Firebolt = 0,
	Healing,
	Lightning,
	Flash,
	Identify,
	FireWall,
	TownPortal,
	StoneCurse,
	Infravision,
	HealOther,
	Nova,
	Fireball,
	ManaShield,
	FlameWave,
	Inferno,
	ChainLightning,
	Sentinel, // unused
	DoomSerpents,
	BloodRitual,  // unused
	Invisibility, // unused
	Golem,
	Etherealize,
	BloodBoil,
	Teleport,
	Apocalypse,
	ItemRepair,
	Phasing = 27,`

Reply:
Like this?

Reply:
IMO, there's nothing to solve. If anything, maybe rename the spell icon to `Null`, `Blank`, or `Empty` instead of `None`.

Reply:
> IMO, there's nothing to solve. If anything, maybe rename the spell icon to `Null`, `Blank`, or `Empty` instead of `None`.

I'll go with `Empty`, same as the enum found in the forbidden code we don't talk about

Reply:
Yes, you are missing some things. That array provides the mapping to look up the spell icon given the spell ID. Therefore, the order in the array corresponds to the order of the values in the `SpellID` enum. The `SpellIcon` enum value is converted to an integer to get the frame index when loading the CEL data for the icon.

Conventional wisdom suggests that you shouldn't rely on the underlying value of the enum, but we'd need to maintain a mapping to look up the frame index regardless. There's no way around it. The `SpellIcon` enum just makes it a bit more readable.

Reply:
Ah fair enough... thanks for elaborating on that and apologies for the disruption here.